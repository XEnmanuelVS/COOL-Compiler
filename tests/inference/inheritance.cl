class A inherits B {
    set_value(new_value: Int): AUTO_TYPE {
        x <- new_value
    };
};

class B inherits C {};

class C inherits D {};

class D inherits E {};

class E {
    x: AUTO_TYPE;
};
