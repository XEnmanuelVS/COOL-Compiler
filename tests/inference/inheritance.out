\__ProgramNode [<class> ... <class>]
  \__ClassDeclarationNode: class A : B { <feature> ... <feature> }
    \__FuncDeclarationNode: set_value(new_value: Int) : Int -> <body>
      \__AssignNode: x <- <expr>
        \__VariableNode: new_value
  \__ClassDeclarationNode: class B : C { <feature> ... <feature> }

  \__ClassDeclarationNode: class C : D { <feature> ... <feature> }

  \__ClassDeclarationNode: class D : E { <feature> ... <feature> }

  \__ClassDeclarationNode: class E  { <feature> ... <feature> }
    \__AttrDeclarationNode: x : Int
